### PR TITLE
ci: logictest support regex with new query type R

### PR DESCRIPTION
**Pull request by ZeaLoVe**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Support regex in logictest, use new query type R. Usage in the readme.md "How to use regex in logic test".
Migrate some stateless tests with *.result_filter.

Fixes #issue

### CODE REVIEW

File: tests/logictest/regex_type.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
+import re
+from log import log
+
+regex_type_map = {
+    'ANYTHING': '.*',  # empty space will make problem
+    'DATE': '\d\d\d\d-\d\d-\d\d \d\d:\d\d:\d\d[.]\d\d\d [+-]\d\d\d\d',
```
Comment (sundy-li) [inline tests/logictest/regex_type.py:6]: \d{4}-\d{2}-\d{2} ...

Comment (ZeaLoVe) [reply tests/logictest/regex_type.py:6]: I just copy from stateless suites ...

**Comment by ZeaLoVe** [comment]
@mergify update

**Review by sundy-li — APPROVED** [review]

**Review by flaneur2020 — APPROVED** [review]

File: tests/logictest/README.md [hunk 2]
```diff
@@ -147,29 +171,29 @@ select 1;
...
-- *warning* A statement query need result, and even if you want to skip a case, you still need to keep the results in the test content
+- *warning* A statement query need results, and even if you want to skip a case, you still need to keep the results in the test content
 
 ## Tools
 
 complete.py can auto complete test file for you, It do as follow steps:
 
 1. Get SQLs from source-file, whether an SQL file or logictest file.
-2. Execute SQL one by one, if SQL fetch result, add statement query; if SQL fetch nothing, add statement ok; if SQL get an error, add statement error
+2. Execute SQL one by one, if SQL fetches results, add statement query; if SQL fetches nothing, add statement ok; if SQL gets an error, add statement error.
```
Comment (flaneur2020) [inline tests/logictest/README.md:183]: fetches -> fetched ?

**Pull request merged**